libsolidity/analysis/ControlFlowGraph.h

```diff
@@ -141,9 +141,11 @@ class CFG: private ASTConstVisitor
 		// containers using this struct
 		bool operator<(FunctionContractTuple const& _other) const
 		{
-			if (contract && _other.contract)
-				if (contract->id() != _other.contract->id())
-					return contract->id() < _other.contract->id();
+			int64_t contractId = contract ? contract->id() : -1;
+			int64_t otherContractId = _other.contract ? _other.contract->id() : -1;
+
+			if (contractId != otherContractId)
+				return contractId < otherContractId;
 
 			return function->id() < _other.function->id();
```

Comment:
```
return
    make_pair(contract ? -1 : contract->id(), function->id()) <
    make_pair(_other.contract ? -1 : _other.contract->id(), _other.function->id());

Reply:
I'm actually wondering if there is a tool somewhere that generates comparison operators based on projections, i.e. something like
```
return KeyComparator{[](FunctionContractTuple const& _x) { return {_x ? -1 : _x->id(), _x.function->id()}}(*this, _other);
```